File: math/generate-random-point-in-a-circle/generate-random-point-in-a-circle.js
/**
 * @param {number} radius
 * @param {number} x_center
 * @param {number} y_center
 */
function Solution(radius, x_center, y_center) {
  this.radius = radius;
  this.x = x_center;
  this.y = y_center;
}

/**
 * @return {number[]}
 */
Solution.prototype.randPoint = function () {
  const len = Math.sqrt(Math.random()) * this.radius;
  const deg = Math.random() * 2 * Math.PI;
  const x = this.x + len * Math.cos(deg);
  const y = this.y + len * Math.sin(deg);

  return [x, y];
};

/**
 * Your Solution object will be instantiated and called as such:
 * var obj = new Solution(radius, x_center, y_center)
 * var param_1 = obj.randPoint()
 */

File: math/generate-random-point-in-a-circle/generate-random-point-in-a-circle.java
class Solution {
  double radius;
  double x;
  double y;

  public Solution(double radius, double x_center, double y_center) {
    this.radius = radius;
    this.x = x_center;
    this.y = y_center;
  }

  public double[] randPoint() {
    double len = Math.sqrt(Math.random()) * radius;
    double deg = Math.random() * 2 * Math.PI;
    double x = this.x + len * Math.cos(deg);
    double y = this.y + len * Math.sin(deg);

    return new double[] { x, y };
  }
}

/**
 * Your Solution object will be instantiated and called as such: Solution obj =
 * new Solution(radius, x_center, y_center); double[] param_1 = obj.randPoint();
 */
